ikarus/solver/nonLinearSolver/solverInfos.hh

```diff
@@ -0,0 +1,13 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+namespace Ikarus {
+  struct NonLinearSolverInformation {
+    explicit operator bool() const { return success; }
+    bool success{false};
+    double norm{0.0};
```

Comment:
This should be generalized to use contain the residualnorm and the correction norm, since both can be valid convergence criteria